### PR TITLE
Update black, isort

### PR DESCRIPTION
**Pull request by smackesey**
Updates `black` (20.8b1 -> 22.1.0) and `isort` (<5.0 > 5.10.1). Also condenses configuration.

Reasons to upgrade black:

- faster
- accounts for Python 3.9, 3.10
- new `--required-version` option prevents annoying snafus where one accidentally formats locally with a different version than Dagster uses
- improved gitignore support (reads gitignore at all levels, like git itself), allowing reducing use of `--exclude` patterns
- now supports Jupyter notebooks (there are quite a few of these in our examples)

Reasons to upgrade isort:

- less duplicate configuration with black bc of new --profile=black option
- doesn't auto-hoist imports, more flexible pragmas (useful in particular for docs_snippets with multiple sections)
- bugfixes

This PR also:

- Gets rid of most of the excluded patterns in black, which are covered by gitignore or black's defaults anyway.
- Removes several CLI options from the `Makefile` black commands. Everything not specific to the formatting target is in the config file `pyproject.toml`. This is good because all invocations of black (for instance, from your editor) will read this config, preventing differences between the `Makefile` invocation and editor invocations.
- Prefixes the black invocations in the `Makefile` with `-`. This tells make to run the next command even if a command exits with a non-0 exit code. E.g. previously, `make check_black` would stop running if the first invocation found any errors, with "-" prefix it continues to the second invocation no matter what.
- Adds a second `pyproject.toml` to `examples/docs_snippets`, since the line length requirement for docs_snippets is different than the rest of the project (see #6703).
- Reformats docs_snippets with the shorter line-length, partially addressing #6703.
- Gets rid of `.isort.cfg` and moves config to `pyproject.toml`
- Makes the paths processed by black and isort identical (previously isort would target snapshots, black wouldn't, now isort ignores them too).
- Makes isort use same line-length as black (previously isort was using 78)
- Pins isort version
- Adjusts buildkite isort step so that:
    - Uses the pinned isort version
    - Calls a new `check_isort` task just like `black` does

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6714** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6714" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: examples/airline_demo/airline_demo/notebooks/Delays_by_Geography.ipynb [hunk 1]
```diff
@@ -22,18 +22,16 @@
     "from dagster import ModeDefinition, ResourceDefinition\n",
     "from collections import namedtuple\n",
     "\n",
-    "url = 'postgresql://{username}:{password}@{hostname}:5432/{db_name}'.format(\n",
-    "    username='test', password='test', hostname='localhost', db_name='test'\n",
+    "url = \"postgresql://{username}:{password}@{hostname}:5432/{db_name}\".format(\n",
+    "    username=\"test\", password=\"test\", hostname=\"localhost\", db_name=\"test\"\n",
```
Comment (alangenfeld) [inline examples/airline_demo/airline_demo/notebooks/Delays_by_Geography.ipynb:26]: these quote changes seem less than ideal, assuming the escapes show up in the notebook when viewed in a tool. 

Do they?

Comment (smackesey) [reply examples/airline_demo/airline_demo/notebooks/Delays_by_Geography.ipynb:26]: Just checked, they don't.

**Review by alangenfeld — APPROVED** [review]
dope

**Pull request merged**